### PR TITLE
[RFR][NOTEST] Refactor EC2 Template Upload for CFME QE

### PR DESCRIPTION
**Pull request by mshriver**
Refactoring the script to be region-aware, to use several new methods in wrapanapi.EC2System, and to cleanup after itself.

Requires wrapanapi v2.1.2, update to frozen.txt included here.

Plenty of TODO's that may or may not get addressed right now. Currently CFME QE is only testing against a single region. A lot of possible improvements to this script are only valid when uploading to multiple regions, and I don't see a need or value to addressing them right now.

### CODE REVIEW

**Comment by dajoRH** [comment]
Lint report for commit e283d4c4c15913a326c3078272322f72fac3ea02:

`scripts/template_upload_ec2.py`:
- :bangbang: Line 168:5: **E265** *block comment should start with '# '*
- :bangbang: Line 169:5: **E265** *block comment should start with '# '*
- :bangbang: Line 248:9: **E265** *block comment should start with '# '*

Please, rectify these issues :smirk: .


**No commit flaws detected**.

*CFME QE Bot*

File: scripts/template_upload_ec2.py [hunk 1]
```diff
@@ -96,143 +110,158 @@ def download_image_file(image_url, destination=None):
...
+    for prov_key in prov_to_upload:
+        region = mgmt_sys[prov_key].get('region', prov_key)
+        upload_bucket_name = mgmt_sys[prov_key].get('upload_bucket_name', 'cfme-upload')
+
+        print('EC2:{}:{} Starting S3 upload of {}'.format(prov_key, region, file_path))
+        ec2_connect_args = {
+            'username': credentials[mgmt_sys[prov_key]['credentials']]['username'],
+            'password': credentials[mgmt_sys[prov_key]['credentials']]['password'],
+            'region': mgmt_sys[prov_key]['region']}
+        ec2 = EC2System(**ec2_connect_args)
```
Comment (mmojzis) [inline scripts/template_upload_ec2.py]: Maybe get_mgmt should be better? To take creds from yamls?

File: scripts/template_upload_ec2.py [hunk 2]
```diff
@@ -96,143 +110,158 @@ def download_image_file(image_url, destination=None):
...
+    """
+    Create a bucket and upload the given file to the bucket
+    :param ec2: mgmtsystem:EC2System object
+    :param upload_bucket_name: string name of bucket to create/upload to
+    :param ami_name: string name of the file in the bucket
+    :param file_path: string path to the local file to upload
+    :return: none
+    """
+    print("EC2:{}: Creating bucket {}...".format(ec2.api.region, upload_bucket_name))
+    ec2.create_s3_bucket(upload_bucket_name)
```
Comment (mmojzis) [inline scripts/template_upload_ec2.py]: You are creating for every upload a bucket? So there should be removal of this bucket after successful import into AMI? I can create method for deleting buckets in wrapanapi.

File: scripts/template_upload_ec2.py (hunk 1 above)
Comment (mshriver) [inline scripts/template_upload_ec2.py]: Definitely, have this in local branch now.

File: scripts/template_upload_ec2.py (hunk 2 above)
Comment (mmojzis) [inline scripts/template_upload_ec2.py]: Also bucket should be emptied first. So I should create method called like force_delete_s3_bucket, which first empties bucket and then deletes it.

Comment (mshriver) [inline scripts/template_upload_ec2.py]: I need to play with the bucket region settings some more, but I think at the moment I need the ability to delete a key from a bucket more than actually deleting the bucket.

wrapanapi:create_s3_bucket returns cleanly if the bucket exists, so this works fine without actually deleting the bucket.

force_delete_s3_bucket that automatically deletes keys first would be very convenient method.

**Comment by dajoRH** [comment]
Lint report for commit be34968d29fe548a5715aee1e64a9520e52bd229:

`scripts/template_upload_ec2.py`:
- :red_circle: Line 21:1: **F401** *'utils.conf.credentials' imported but unused*
- :bangbang: Line 136:5: **E265** *block comment should start with '# '*
- :bangbang: Line 138:5: **E265** *block comment should start with '# '*
- :bangbang: Line 141:5: **E265** *block comment should start with '# '*
- :bangbang: Line 143:5: **E265** *block comment should start with '# '*
- :bangbang: Line 172:5: **E265** *block comment should start with '# '*
- :bangbang: Line 173:5: **E265** *block comment should start with '# '*
- :bangbang: Line 239:9: **E265** *block comment should start with '# '*
- :bangbang: Line 243:9: **E265** *block comment should start with '# '*

Hmmm
:frowning: You might like to check the lint of your code :point_up: 


**No commit flaws detected**.

*CFME QE Bot*

**Comment by dajoRH** [comment]
Would you mind rebasing this Pull Request against latest master, please? :trollface:
*CFME QE Bot*

**Pull request merged**